server/shadow/shadow_client.c

```diff
@@ -938,7 +939,8 @@ void* shadow_client_thread(rdpShadowClient* client)
 	peer->update->SurfaceFrameAcknowledge = (pSurfaceFrameAcknowledge) shadow_client_surface_frame_acknowledge;
 
 	StopEvent = client->StopEvent;
-	UpdateEvent = subsystem->updateEvent;
+	UpdateSubscriber = shadow_multiclient_get_subscriber(subsystem->updateEvent);
```

Comment:
shadow_multiclient_get_subscriberreturns is not checked for NULL ? Should the thread run in that case ?